.travis.yml

```diff
@@ -1,33 +1,66 @@
 language: cpp
 sudo: false
-compiler: clang
-#    - clang
-#    - gcc
+compiler:
+    - clang
+    - gcc
 
-os: osx
-#    - linux
-#    - osx
+os:
+    - linux
+    - osx
+#osx_image: xcode7
+dist: trusty
+
+addons:
+  apt: 
+   sources:
+      - boost-latest
+      - ubuntu-toolchain-r-test
+   packages:
+      - gcc-4.8
+      - g++-4.8
+      - libboost1.55-all-dev
+      - libilmbase-dev
+      - libopenexr-dev
+      - libgif-dev
+      - libtiff4-dev
+      - libopenjpeg-dev
+      - libwebp-dev
+      - ffmpeg
```

Comment:
Don't think for a minute that the docs were clear about any of this!

It took a lot of hunting through docs (which are in many cases outdated), reading their blog posts that explain things not spelled out in the docs (especially for recent changes), and hunting around the net for other C++-based projects that use Travis and seeing what their .travis.yml files looked like.

And I've been through 98 builds so far, figuring this out by trial and error!